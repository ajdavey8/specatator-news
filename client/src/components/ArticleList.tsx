import React from 'react';
import styled from 'styled-components';
import { Article } from '../types';
import ArticleListItem from './ArticleListItem';

const Container = styled.section`
  margin: ${({ theme }) => theme.spacingM};

  ${({ theme }) => theme.mediaMedium} {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    column-gap: ${({ theme }) => theme.spacingXL};
  }

  ${({ theme }) => theme.mediaLarge} {
    padding-right: ${({ theme }) => theme.spacingXL};
    border-right: 0.5px solid ${({ theme }) => theme.lightGrey};
    border-image: linear-gradient(to bottom, ${({ theme }) => theme.lightGrey} 0%, ${({ theme }) => theme.lightGrey} 94%,rgba(38, 34, 34,0) 94%,rgba(38, 34, 34,0) 100%);
    border-image-slice: 1;
  }

`

interface ArticleList {
  articles: Article[]
  addBookmark: (arg1: number) => void
}

const ArticleList = ({ articles, addBookmark }: ArticleList) => (
  <Container>
    {articles?.map(article => (
      <ArticleListItem {...article} key={article.id} addBookmark={addBookmark} />
    ))}
  </Container>
)

export default ArticleList;
