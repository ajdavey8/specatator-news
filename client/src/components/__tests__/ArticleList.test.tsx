
import React from 'react';
import { render } from '@testing-library/react'
import ArticleList from '../ArticleList'
import { ThemeProvider } from 'styled-components';

const addBookmark = jest.fn()

const theme = {
  green: "#29724c",
}

const articles = [
  {
    id: 1,
    title: 'The magic and mystery of English Cricket',
    author: 'Alex Massie',
    image: 'image'
  },
  {
    id: 2,
    title: 'Starman was firing blanks at PMQs',
    author: 'Lloyd Evans',
    image: 'image'
  },
]

const setup = () => {
  const utils = render(
    <ThemeProvider theme={theme}>
      <ArticleList
        articles={articles}
        addBookmark={addBookmark}
      />
    </ThemeProvider>
  )

  return {
    ...utils
  }
}

describe('<ArticleList>', () => {
  it('should return articles', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})