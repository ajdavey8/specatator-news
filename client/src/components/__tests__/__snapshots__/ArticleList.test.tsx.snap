// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList> should return articles 1`] = `[Function]`;
